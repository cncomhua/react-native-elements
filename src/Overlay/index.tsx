import { withTheme } from '../config';
import { Overlay, OverlayProps } from './Overlay';

export { Overlay };
export type { OverlayProps };
/**
 * RNE Overlay
 *
 * **Preview**
 *
 * ![URL](URL)
 *
 * [**Documentation**](https://reactnativeelements.com/docs/overlay)
 */
export default withTheme(Overlay, 'Overlay');
