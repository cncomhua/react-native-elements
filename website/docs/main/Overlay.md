---


id: overlay


title: Overlay


slug: /overlay

---



import Usage from './usage/Overlay/Overlay.md'



The Overlay is a view that floats above an appâ€™s content.

Overlays are an easy way to inform or request information from the user.



## Usage


<Usage />

---


## Props

### Overlay




## Reference

### Overlay

