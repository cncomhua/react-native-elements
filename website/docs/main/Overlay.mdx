---
id: overlay

title: Overlay

slug: /overlay
---

import Usage from './usage/Overlay/Overlay.mdx';

import Props from './props/Overlay.mdx';

import { Overlay } from 'react-native-elements';

import Tabs from '@theme/Tabs';

import TabItem from '@theme/TabItem';

The Overlay is a view that floats above an appâ€™s content.

Overlays are an easy way to inform or request information from the user.

## Usage

<Usage />

## Props

<Props />
