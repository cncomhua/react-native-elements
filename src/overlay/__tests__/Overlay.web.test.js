import React from 'react';
import { Text, TouchableHighlight, View } from 'react-native';
import foreach from 'lodash.foreach';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import parseSamples, {
  assignRef,
  genRefId,
} from 'enzyme-styleguidist-sample-parser';
import options, {
  snapShot,
  buildJsxForGuideMethod,
  ensureCalled,
  onlyEnsureCalled,
  onlySnapshots,
} from '../../../samples';
import Overlay from '../Overlay';

const children = <Text>here's something</Text>;
const childrenString = `<Text>here's something</Text>`;

const getJsx = propString => {
  const jsx = `initialState = { isVisible: false };
const View = RN.View;
const TouchableHighlight = RN.TouchableHighlight;
const Text = RN.Text;
    
<View>
  <TouchableHighlight
    onPress={() => {
      setState({ isVisible: true });
      setTimeout(() => {
        setState({ isVisible: false });
      }, 4000);
    }}
    style={{ backgroundColor: '#aaa', padding: 10, marginBottom: 15 }}
  >
    <Text>Show Overlay for 4 sec</Text>
  </TouchableHighlight>
  <Overlay ${propString} isVisible={state.isVisible}>
    ${childrenString}
  </Overlay>
</View>
`;
  console.warn(jsx);
  return jsx;
};

// const props = {
//   containerStyle: {
//     component: Overlay,
//     children,
//     props: {
//       containerStyle: { borderWidth: 5 },
//     },
//     enzyme: {
//       tests: { shallow: { snapshot: snapShot() } },
//     },
//     styleguidist: {
//       getJsxString: getJsx('containerStyle={{ borderWidth: 5 }}'),
//     },
//   },
//   overlayStyle: {
//     component: Overlay,
//     children,
//     props: {
//       overlayStyle: { borderWidth: 5 },
//     },
//     enzyme: {
//       tests: { shallow: { snapshot: snapShot() } },
//     },
//     styleguidist: {
//       getJsxString: getJsx('overlayStyle={{ borderWidth: 5 }}'),
//     },
//   },
// };

const props = {
  containerStyle: {
    component: Overlay,
    children,
    props: {
      isVisible: true,
      containerStyle: { borderWidth: 5 },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  overlayStyle: {
    component: Overlay,
    children,
    props: {
      isVisible: true,
      overlayStyle: { borderWidth: 5 },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  overlayBackgroundColor: {
    component: Overlay,
    children,
    props: {
      isVisible: true,
      overlayBackgroundColor: '#aaa',
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  windowBackgroundColor: {
    component: Overlay,
    children,
    props: {
      isVisible: true,
      windowBackgroundColor: '#aaa',
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  borderRadius: {
    component: Overlay,
    children,
    props: {
      isVisible: true,
      borderRadius: 20,
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  width: {
    component: Overlay,
    children,
    props: {
      isVisible: true,
      width: 50,
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  height: {
    component: Overlay,
    children,
    props: {
      isVisible: true,
      height: 50,
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  fullScreen: {
    component: Overlay,
    children,
    props: {
      isVisible: true,
      fullScreen: true,
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
};

const samples = {
  props,
};

parseSamples(
  {
    Misc: { sectionComponents: { Overlay: { samples } } },
  },
  options
);
