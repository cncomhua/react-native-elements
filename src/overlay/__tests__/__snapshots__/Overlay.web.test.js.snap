// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Misc: Overlay: Props Misc: Overlay: [ prop: borderRadius ] enzyme-shallow: snapshot: Misc: Overlay: [ prop: borderRadius ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(0, 0, 0, .4)",
      "height": 768,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "top": 0,
      "width": 1024,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 20,
        "height": 588,
        "padding": 10,
        "width": 944,
      }
    }
  >
    <Text>
      here's something
    </Text>
  </View>
</View>
`;

exports[`Misc: Overlay: Props Misc: Overlay: [ prop: containerStyle ] enzyme-shallow: snapshot: Misc: Overlay: [ prop: containerStyle ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(0, 0, 0, .4)",
      "borderWidth": 5,
      "height": 768,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "top": 0,
      "width": 1024,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 3,
        "height": 588,
        "padding": 10,
        "width": 944,
      }
    }
  >
    <Text>
      here's something
    </Text>
  </View>
</View>
`;

exports[`Misc: Overlay: Props Misc: Overlay: [ prop: fullScreen ] enzyme-shallow: snapshot: Misc: Overlay: [ prop: fullScreen ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(0, 0, 0, .4)",
      "height": 768,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "top": 0,
      "width": 1024,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 3,
        "height": 768,
        "padding": 10,
        "width": 1024,
      }
    }
  >
    <Text>
      here's something
    </Text>
  </View>
</View>
`;

exports[`Misc: Overlay: Props Misc: Overlay: [ prop: height ] enzyme-shallow: snapshot: Misc: Overlay: [ prop: height ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(0, 0, 0, .4)",
      "height": 768,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "top": 0,
      "width": 1024,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 3,
        "height": 50,
        "padding": 10,
        "width": 944,
      }
    }
  >
    <Text>
      here's something
    </Text>
  </View>
</View>
`;

exports[`Misc: Overlay: Props Misc: Overlay: [ prop: overlayBackgroundColor ] enzyme-shallow: snapshot: Misc: Overlay: [ prop: overlayBackgroundColor ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(0, 0, 0, .4)",
      "height": 768,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "top": 0,
      "width": 1024,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#aaa",
        "borderRadius": 3,
        "height": 588,
        "padding": 10,
        "width": 944,
      }
    }
  >
    <Text>
      here's something
    </Text>
  </View>
</View>
`;

exports[`Misc: Overlay: Props Misc: Overlay: [ prop: overlayStyle ] enzyme-shallow: snapshot: Misc: Overlay: [ prop: overlayStyle ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(0, 0, 0, .4)",
      "height": 768,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "top": 0,
      "width": 1024,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 3,
        "borderWidth": 5,
        "height": 588,
        "padding": 10,
        "width": 944,
      }
    }
  >
    <Text>
      here's something
    </Text>
  </View>
</View>
`;

exports[`Misc: Overlay: Props Misc: Overlay: [ prop: width ] enzyme-shallow: snapshot: Misc: Overlay: [ prop: width ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(0, 0, 0, .4)",
      "height": 768,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "top": 0,
      "width": 1024,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 3,
        "height": 588,
        "padding": 10,
        "width": 50,
      }
    }
  >
    <Text>
      here's something
    </Text>
  </View>
</View>
`;

exports[`Misc: Overlay: Props Misc: Overlay: [ prop: windowBackgroundColor ] enzyme-shallow: snapshot: Misc: Overlay: [ prop: windowBackgroundColor ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#aaa",
      "height": 768,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "top": 0,
      "width": 1024,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 3,
        "height": 588,
        "padding": 10,
        "width": 944,
      }
    }
  >
    <Text>
      here's something
    </Text>
  </View>
</View>
`;
